app/models/agents/ftpsite_agent.rb

```diff
@@ -176,17 +182,25 @@ def saving_entries
       new_files.sort_by { |filename|
         found_entries[filename]
       }.each { |filename|
-        create_event payload: {
+        create_event payload: get_file_pointer(filename).merge({
           'url' => (base_uri + uri_path_escape(filename)).to_s,
           'filename' => filename,
           'timestamp' => found_entries[filename],
-        }
+        })
       }
 
       memory['known_entries'] = found_entries
       save!
     end
 
+    def get_io(file)
```

Comment:
It does, `Net::FTP` can yield every line to a block when reading text files, to not buffer the contents in RAM we would need to write a wrapper around it which more or less implements the `IO` class. That would probably involve some threading or fibers. I figured for the initial version it would be good enough to read the whole file into RAM.